xbmc/platform/android/CPUInfoAndroid.h

```diff
@@ -8,17 +8,26 @@
 
 #pragma once
 
+#include "utils/CPUInfo.h"
 #include "utils/Temperature.h"
 
-#include "platform/posix/CPUInfoPosix.h"
+#include "platform/posix/filesystem/PosixFile.h"
 
-class CCPUInfoAndroid : public CCPUInfoPosix
```

Comment:
Is there a reason to not use `CCPUInfoPosix`?

That class provides a fallback for reading a script from advancedsettings to allow a user to provide an override for the cpu temperature.

See, https://github.com/xbmc/xbmc/blob/ef37e9abaad7c8a188c4656e6b89499b5caecf4a/xbmc/platform/posix/CPUInfoPosix.cpp#L15

Reply:
Yes, I have two reasons:

1) Eliminate wrong/nonsense temperature indication (-1 ºC).

2) Fallback script can't really be used on Android unless the device is rooted/unlocked.

3) 99.999% of people who use Android on Fire TV Stick or similar they are not interested (I guess) in writing a script to get the CPU temperature (even if could be used on rooted devices). On the contrary, there may be a significant percentage of people who worry and think that something is wrong if they only see empty fields and a temperature of -1 ºC. Because of this, there are even people who care enough to open an Github issue... 😉